plugin/src/index.ts

```diff
@@ -16,7 +16,7 @@ import {
   shouldSkipBundlingDatastore,
 } from './helpers/config'
 import { modifyFiles } from './helpers/files'
-import { deleteFunctions, writeFunctions } from './helpers/functions'
+import { deleteFunctions } from './helpers/functions'
```

Comment:
I think we'll still have the same problem here because `import`ing this module will mean it has to resolve any dependencies of the module. That will still attempt to pull in `@gatsby/reach-router` won't it?

Reply:
Using a dynamic `import()` means that it'll only try to load it in that code path

Reply:
Oh wait, I missed the `deleteFucntions` path! Good point. Have you tested this on a site that doesn't have the module installed? It'll need to be tested with a manual install in a separate site, because otherwise dependencies in the monorepo will mean that it could still resolve it even if the site uses Gatsby 2.

Reply:
Yeah - I don't think it works with the current implementation due to the extra import at the top.

I think @taty2010 has a test repo which will fail. https://github.com/taty2010/gatsby-v2-test
However, this adds the `@gatsby/reach-router` dependency. If we remove that, I think it'll fail.

Reply:
Actually, looking again at the code it's not actually importing `@gatsby/reach-router` in `./helpers/functions`. It's only imported in the generated API function itself. What's unclear is why this is being created for those sites, because it's only supposed to be generated if there are Gatsby functions defined.

Reply:
@nickytonline and I tested out the update locally using the Gatsby v2 test site and made sure we uninstalled the `@gatsby/reach-router` beforehand. I'll dig into it everything that was brought up, thanks!

Reply:
@ascorbic - it's still going to attempt to resolve `@gatsby/reach-router` isn't it? It imports it from `gatsbyFunction.ts` from `handlers.ts` -> `functions.ts` -> `index.ts`. It's ultimately only used at runtime but it'll still have to resolve the module import.

Reply:
@MarcL comments seem accurate. In my testing I still get

```
Error message
  Could not import plugin:
  Error: Cannot find module '@gatsbyjs/reach-router'
  Require stack:
  - /home/misiek/test/npg-pr557/node_modules/@netlify/plugin-gatsby/lib/templates/api/gatsbyFunction.js
  - /home/misiek/test/npg-pr557/node_modules/@netlify/plugin-gatsby/lib/templates/handlers.js
  - /home/misiek/test/npg-pr557/node_modules/@netlify/plugin-gatsby/lib/helpers/functions.js
  - /home/misiek/test/npg-pr557/node_modules/@netlify/plugin-gatsby/lib/index.js
 ```

https://www.diffchecker.com/160Do9ad/ is how this PR changes built package/module and there it still has all the same `require` calls, as `latest`. `writeFunctions` being moved to dynamic import result in things being added on top of `latest` code and doesn't seem to fully address the import issue yet.

Reply:
This comment thread is no longer relevant. 

PR implementation was changed to just not even try to import/require `@gatsbyjs/reach-router` during the build and it only is ever resolved now when `zip-it-and-ship-it` tries to bundle functions. And gatsby functions wrapper is only generated when gatsby version support it (so since `gatsby@3.7.0`) and all those gatsby versions do come with `@gatsbyjs/reach-router` (gatsby switched to it in `gatsby@3.0.0`)